fuel/datasets/hdf5.py

```diff
@@ -146,23 +165,55 @@ def __init__(self, file_or_path, which_set, subset=None,
         else:
             self.path = file_or_path
             self.external_file_handle = None
-        self.driver = driver
-        self.sort_indices = sort_indices
-        if which_set not in self.available_splits:
-            raise ValueError(
-                "'{}' split is not provided by this ".format(which_set) +
-                "dataset. Available splits are " +
-                "{}.".format(self.available_splits))
+        if not isinstance(which_set, (list, tuple)):
+            which_set = (which_set,)
         self.which_set = which_set
         subset = subset if subset else slice(None)
-        if subset.step not in (1, None):
+        if hasattr(subset, 'step') and subset.step not in (1, None):
             raise ValueError("subset.step must be either 1 or None")
         self._subset_template = subset
         self.load_in_memory = load_in_memory
+        self.driver = driver
+        self.sort_indices = sort_indices
+
+        self._parse_dataset_info()
 
-        kwargs.setdefault('axis_labels', self.load_axis_labels())
+        kwargs.setdefault('axis_labels', self.default_axis_labels)
         super(H5PYDataset, self).__init__(**kwargs)
 
+    def _parse_dataset_info(self):
+        """Parses information related to the HDF5 interface.
+
+        In addition to verifying that the `self.which_set` split is
+        available, this method sets the following attributes:
+
+        * `provides_sources`
+        * `vlen_sources`
+        * `default_axis_labels`
+
+        """
+        self._out_of_memory_open()
+        handle = self._file_handle
+        available_splits = self.get_all_splits(handle)
+        which_set = self.which_set
+        provides_sources = None
+        for split in which_set:
+            if split not in available_splits:
+                raise ValueError(
+                    "'{}' split is not provided by this ".format(split) +
+                    "dataset. Available splits are " +
+                    "{}.".format(available_splits))
+            split_provides_sources = set(
+                self.get_provided_sources(handle, split))
+            if provides_sources:
+                provides_sources &= split_provides_sources
+            else:
+                provides_sources = split_provides_sources
+        self.provides_sources = tuple(sorted(provides_sources))
```

Comment:
I am trying to find where these attributes were set in the previous version of the code but so far I fail. Can you please help?